deps/src/singular.cpp

```diff
@@ -53,6 +75,46 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     singular_define_ideals(Singular);
     singular_define_matrices(Singular);
 
+    Singular.method("call_interpreter", [](std::string input) {
+        // set callbacks
+        auto default_print = PrintS_callback;
+        auto default_error = WerrorS_callback;
+        auto default_warning = WarnS_callback;
+
+        PrintS_callback = PrintS_for_julia;
+        WerrorS_callback = WerrorS_for_julia;
+        WarnS_callback = WarningS_for_julia;
+
+        // cleanup return strings
+        singular_return.erase();
+        singular_error.erase();
+        singular_warning.erase();
```

Comment:
Huh, I didn't know `erase()` works for `std::string` (but then, my C++ is rather rusty). Back in the day, it was `clear()`.

Reply:
`erase` can actually take arguments which part of the string is deleted. With no arguments, it behaves like `clear`